cpp/daal/include/services/internal/sycl/math/mkl_blas.h

```diff
@@ -120,19 +160,42 @@ struct MKLSyrk
     {
         Status status;
 
-        const MKL_TRANSPOSE transmkl = trans == math::Transpose::Trans ? MKL_TRANS : MKL_NOTRANS;
-        const MKL_UPLO uplomkl       = upper_lower == math::UpLo::Upper ? MKL_UPPER : MKL_LOWER;
+#ifdef DAAL_SYCL_INTERFACE_USM
+        if (a_buffer.isUSMBacked() && c_buffer.isUSMBacked())
+        {
+            const auto transmkl = to_fpk_transpose(trans);
+            const auto uplomkl  = to_fpk_uplo(upper_lower);
+
+            auto a_usm = a_buffer.toUSM(status);
+            DAAL_CHECK_STATUS_VAR(status);
+            auto c_usm = c_buffer.toUSM(status);
+            DAAL_CHECK_STATUS_VAR(status);
+
+            auto a_ptr = a_usm.get() + offsetA;
+            auto c_ptr = c_usm.get() + offsetC;
+
+            status |= catchSyclExceptions([&]() mutable {
+                ::oneapi::fpk::blas::syrk(_queue, uplomkl, transmkl, n, k, alpha, a_ptr, lda, beta, c_ptr, ldc);
+                _queue.wait_and_throw();
+            });
+        }
+        else
+#endif
+        {
+            const MKL_TRANSPOSE transmkl = trans == math::Transpose::Trans ? MKL_TRANS : MKL_NOTRANS;
+            const MKL_UPLO uplomkl       = upper_lower == math::UpLo::Upper ? MKL_UPPER : MKL_LOWER;
```

Comment:
```suggestion
            const MKL_UPLO uplomkl       = to_fpk_uplo(uplo);
```

Reply:
Cannot use `to_fpk_uplo` here since it returns `::oneapi::fpk::uplo`, but not a C-constant `MKL_UPLO`